desc/utils.py

```diff
@@ -742,6 +742,15 @@ def atleast_nd(ndmin, ary):
     return jnp.array(ary, ndmin=ndmin) if jnp.ndim(ary) < ndmin else ary
 
 
+def map2(fun, xs, *, batch_size=None):
```

Comment:
no i'm just saying that imap has batched vmap built in and we don't use it. In either case I'd keep this loop and instead do batched imap elsewhere in the bounce integral logic

Reply:
@dpanici is https://github.com/PlasmaControl/DESC/pull/1052/files#r1777684537 always true? Even if it's reduced immediately? Asking whether I should use:

```python
def inner_fun(x):
  x["pitch weight"] = pitch
  x["pitch_inv weight"] = pitch_inv_weight
  return imap(fun, x)

  return imap(
      inner_fun,
      xs,
  )
```
instead of broadcast in _get_pitch_inv_quad?

Reply:
I am not certain if it is always true, tbh that comment is from the JAX source, not one that I made myself. Could be worth trying both and checking peak memory use for a large case?

Reply:
I'm surprised #1288 exists in the first place... but this is the only change made since September that I don't immediately rule out affects JIT